core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.cpp

```diff
@@ -100,11 +103,23 @@ void OpenMPTargetExec::resize_scratch(int64_t team_size, int64_t shmem_size_L0,
   const int64_t shmem_size =
       shmem_size_L0 + shmem_size_L1;  // L0 + L1 scratch memory per team.
   const int64_t padding = shmem_size * 10 / 100;  // Padding per team.
+
+  // Maximum active teams possible.
+  // The number should not exceed the maximum in-flight teams possible or the
+  // league_size.
+  int max_active_teams =
+      std::min(OpenMPTargetExec::MAX_ACTIVE_THREADS / team_size, league_size);
+
+  // max_active_teams is the number of active teams on the given hardware.
+  // We set the number of teams to be twice the number of max_active_teams for
+  // the compiler to pick the right number in its case.
```

Comment:
I do not understand this comment.  Why are we setting the number to twice the max instead of just once?
Also what do you mean "in its case"?

Reply:
The ideal case would be to set an large enough upper bound on the number of teams generated using `omp_set_num_teams` and and let compiler pick up the right number of teams for a give `target` region. We do that in `resize_scratch` where we assign the upper bound to be 2*max_active_teams . However that call is not respected and hence the need to add `num_teams`. 

The idea is to  not hamper compiler's ability to chose the appropriate number of teams (hence a large upper bound) but also to have control over that number so we can allocate data for each team.